deployment/scripts/upgrade.sh

```diff
@@ -0,0 +1,375 @@
+        sleep 1
+      else
+        FOUND=1
+        break
+      fi
+    done
+    if [[ $FOUND -eq 0 ]]; then
+      echo "Unable to find log message from gateway that ES recovered its indices within $TIMES seconds"
+      return 1
+    fi
+  else
+    # Does this check make sense to happen after we check if the node started?
+    # TODO: verify the order that a non-master node does its checks...
+    if [[ -z "$non_master" ]]; then
+      echo "Node isn't master and was unable to detect master"
+      return 1
+    fi
+  fi
+}
+
+function checkKibanaStarted() {
+
+  local pod=$1
+
+  #TODO: this looping logic can probably be refactored into a function
+  FOUND=0
+  for i in $(seq 1 $TIMES); do
+    if [[ -z $(oc logs $1 -c kibana | grep "Listening on 0.0.0.0:5601") ]]; then
+      sleep 1
+    else
+      FOUND=1
+      break
+    fi
+  done
+  if [[ $FOUND -eq 0 ]]; then
+    echo "Kibana was not able to start up within $TIMES seconds"
+    return 0
+  fi
+}
+
+function scaleDown() {
+
+  # check for a fluentd dc, if there isn't one it was either deleted or using a daemonset
+  local fluentd_dc=(`oc get dc -l logging-infra=fluentd -o jsonpath='{.items[*].metadata.name}'`)
+
+  # find what nodes fluentd is currently deployed to
+  fluentd_nodes=(`oc get pods -l component=fluentd -o jsonpath='{.items[*].spec.nodeName}'`)
+  if [[ -z "$fluentd_dc" ]]; then
+    # oc get daemonset -l logging-infra=fluentd -o jsonpath='{.items[*].metadata.name}'
+    local selector=$(oc get daemonset -l logging-infra=fluentd -o jsonpath='{.items[*].spec.template.spec.nodeSelector}')
+    if [[ -n "$selector" ]]; then
+      fluentd_nodeselector=$(echo ${selector:4:-1} | sed 's/:/=/g')
+
+      oc delete daemonset logging-fluentd
+    fi
+  else
+    # we are using a deployment config, scale down here
+    for dc in "${fluentd_dc[@]}"; do
+      oc scale --replicas=0 dc $dc
+    done
+  fi
+
+  for pod in $(oc get pods -l component=fluentd -o jsonpath='{.items[*].metadata.name}'); do
+    waitForStop $pod "component=fluentd"
+  done
+
+  # probably can refactor this out into a function...
+  local kibana_dc=(`oc get dc -l logging-infra=kibana -o jsonpath='{.items[*].metadata.name}'`)
+  for dc in "${kibana_dc[@]}"; do
+    oc scale --replicas=0 dc $dc
+  done
+  for pod in $(oc get pods -l component=kibana -o jsonpath='{.items[*].metadata.name}'); do
+    waitForStop $pod "component=kibana"
+  done
+
+  local elasticsearch_dc=(`oc get dc -l logging-infra=elasticsearch -o jsonpath='{.items[*].metadata.name}'`)
+  for dc in "${elasticsearch_dc[@]}"; do
+    oc scale --replicas=0 dc $dc
+  done
+  for pod in $(oc get pods -l component=es -o jsonpath='{.items[*].metadata.name}'); do
+    waitForStop $pod "component=es"
+  done
+}
+
+function scaleUp() {
+
+  # scale up ES slowly -- we want to wait for each instance to recover its indices first
```

Comment:
i'll experiment with this in the latest commit so we have starts and stops more as the elastic docs state to do their major version upgrades.